crates/driver/src/boundary/settlement.rs

```diff
@@ -238,117 +234,109 @@ fn into_boundary_order(order: &competition::auction::Order) -> Order {
     }
 }
 
-fn into_boundary_solution(solution: competition::Solution) -> SettledBatchAuctionModel {
-    SettledBatchAuctionModel {
+async fn to_boundary_solution(
+    solution: &competition::Solution,
+    eth: &Ethereum,
+) -> Result<SettledBatchAuctionModel> {
+    Ok(SettledBatchAuctionModel {
         orders: solution
-            .orders
-            .into_iter()
-            .map(|(index, order)| {
-                (
+            .trades
+            .iter()
+            .enumerate()
+            .filter_map(|(index, trade)| match trade {
+                competition::solution::Trade::Fulfillment(fulfillment) => Some((
                     index,
                     ExecutedOrderModel {
-                        exec_sell_amount: order.sell,
-                        exec_buy_amount: order.buy,
+                        exec_sell_amount: fulfillment.order.sell.amount,
+                        exec_buy_amount: fulfillment.order.buy.amount,
```

Comment:
This needs to be computed from the `fulfillment.executed` value, and not the order amounts.